### PR TITLE
[XLA] Use tensorflow::port::Aligned{Malloc,Free}

### PR DESCRIPTION
**Pull request by rongjiecomputer**
Split from #15310.

#15213

### CODE REVIEW

**Review by tatatodd — COMMENTED** [review]
This seems fine to me, but let's see what Sanjoy thinks.

File: tensorflow/compiler/xla/service/cpu/external_constant_pool.h [hunk 1]
```diff
@@ -52,7 +53,7 @@ class ExternalConstantPool {
   // We need to `free()` pointers allocated into `entries_` since we allocate
```
Comment (sanjoy) [inline tensorflow/compiler/xla/service/cpu/external_constant_pool.h]: Please update this comment.

**Review by sanjoy — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/cpu/external_constant_pool.h (hunk 1 above)
Comment (rongjiecomputer) [reply tensorflow/compiler/xla/service/cpu/external_constant_pool.h]: Done.

**Review by sanjoy — APPROVED** [review]
LGTM

**Comment by tatatodd** [comment]
Jenkins, test this please.

**Pull request merged**